src/index.ts

```diff
@@ -35,9 +35,20 @@ function init(modules: {
           return;
         }
         if (settings.enable) {
-          prior.definitions = prior.definitions?.filter(({ fileName }) => {
-            const matches = settings.modules.some((t) => fileName.includes(t));
-            if (matches) return false;
+          prior.definitions = prior.definitions?.filter(({ fileName, textSpan, kind, name }) => {
+            if (kind === 'index' && name === '__index') {
+              const definitionNode = findNodeAtPosition(ts, info.languageService.getProgram()!.getSourceFile(fileName)!, textSpan.start)
+              let moduleDeclaration: ts.ModuleDeclaration | undefined
+              ts.findAncestor(definitionNode, node => {
+                if (ts.isModuleDeclaration(node)) {
```

Comment:
So far , this method filter out results  from nextjs declaration and react-scripts declaration files.

However for vite based apps , the definition result points to this [location](https://github.com/vitejs/vite/blob/main/packages/vite/client.d.ts#L4) in the declaration file.

So this specific node in the AST doesn't have any Module Declaration ancestor hence 
```ts
if (ts.isModuleDeclaration(node)) {

}
```
this block never gets executed.

Do you have any other ideas to solve this ?

Reply:
Interesting, didn't know of these cases, and have no ideas for clean solution.

> Do you have any other ideas to solve this ?

Maybe we can include some additional pattern checks as in case of Vite these names are straightforwad:

```ts
  if (kind === ts.ScriptElementKind.indexSignatureElement && name === '__index') {
              if (containerName === 'CSSModule' || containerName === 'CSSModuleClasses') return false
              // ... findNodeAtPosition
```

Reply:
> Interesting, didn't know of these cases, and have no ideas for clean solution.
In theory a library author could emit their declarations however they want.

> Maybe we can include some additional pattern checks as in case of Vite these names are straightforwad:

Can try this out